### PR TITLE
feat(cat-voices): Workspace - more data 837

### PR DESCRIPTION
**Pull request by digitalheartxs**
# Description

Implement more data in Workspace

## Related Issue(s)

Closes https://github.com/input-output-hk/catalyst-voices/issues/837

## Screenshots

https://github.com/user-attachments/assets/8b782a6d-8289-4dc0-956d-a07263b3a742

## Please confirm the following checks

* [x] My code follows the style guidelines of this project
* [x] I have performed a self-review of my code
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation
* [x] My changes generate no new warnings
* [x] I have added tests that prove my fix is effective or that my feature works
* [x] New and existing unit tests pass locally with my changes
* [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

File: catalyst_voices/analysis_options.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 include: package:catalyst_analysis/analysis_options.yaml
 
 analyzer:
-  exclude: [build/**, lib/*.g.dart, lib/generated/**]
+  exclude: [build/**, lib/*.g.dart, lib/generated/**, lib/pages/workspace/rich_text/**]
```
Comment (dtscalac) [inline catalyst_voices/analysis_options.yaml:4]: Please just disable lints per file, i.e. `ignore_for_file: ...`

File: catalyst_voices/lib/pages/workspace/proposal_details.dart [hunk 2]
```diff
@@ -121,43 +120,50 @@ class _StepDetails extends StatelessWidget {
...
-            content: desc!,
-          )
-        : WorkspaceTileContainer(
-            isSelected: isSelected,
-            content: VoicesRichText(
-              title: name,
-              document: doc,
+    if (desc != null) {
+      return WorkspaceTextTileContainer(
+        name: titleInDetails != null ? titleInDetails! : title,
```
Comment (dtscalac) [inline catalyst_voices/lib/pages/workspace/proposal_details.dart:143]: if `titleInDetails` overrides the `title`, can we just set `title` to whatever is in `titleInDetails`?

File: catalyst_voices/lib/pages/workspace/rich_text/problem_statement.dart [hunk 3]
```diff
@@ -0,0 +1,6 @@
+const problemStatement = [
```
Comment (dtscalac) [inline catalyst_voices/lib/pages/workspace/rich_text/problem_statement.dart:1]: consider to move all these samples in a subfolder of `rich_text` folder

File: catalyst_voices/lib/pages/workspace/proposal_details.dart (hunk 2 above)
Comment (digitalheartxs) [reply catalyst_voices/lib/pages/workspace/proposal_details.dart:143]: I don't understand what do you mean.
Do you know what `titleInDetails` is for?

Screenshot 2024-10-25 at 13 28 04

File: catalyst_voices/lib/pages/workspace/rich_text/problem_statement.dart (hunk 3 above)
Comment (digitalheartxs) [reply catalyst_voices/lib/pages/workspace/rich_text/problem_statement.dart:3]: What do you mean? They are currently in subdirectory `rich_text`

Comment (dtscalac) [reply catalyst_voices/lib/pages/workspace/rich_text/problem_statement.dart:3]: The idea is to separate "samples" from "code", i.e. in rich_text you can place the widgets and their utils, in `rich_text/samples` you can add the json samples.

File: catalyst_voices/packages/catalyst_voices_models/lib/src/workspace/workspace_proposal_segment_step.dart [hunk 4]
```diff
@@ -4,27 +4,43 @@ import 'package:equatable/equatable.dart';
...
     required this.title,
+    this.titleInDetails,
     this.description,
     this.documentJson,
+    this.richTextParams,
     this.isEditable = false,
   }) : assert(
-          description != null || documentJson != null,
+          description != null || richTextParams != null,
           'Make sure description or document are provided',
```
Comment (dtscalac) [inline catalyst_voices/packages/catalyst_voices_models/lib/src/workspace/workspace_proposal_segment_step.dart:23]: Please update assert message to reflect the logic inside the assert

File: catalyst_voices/lib/pages/workspace/proposal_details.dart (hunk 2 above)
Comment (dtscalac) [reply catalyst_voices/lib/pages/workspace/proposal_details.dart:143]: I was suggesting to simplify the logic by extracting the decision whether to use the `title` or `titleInDetails` to the parent widget. `_StepDetails` could just have a single parameter called title, and you would feed it like:

```
      return _StepDetails(
                key: ValueKey('WorkspaceStep${step.id}TileKey'),
                id: step.id,
                title: step.titleInDetails ?? step.title,
                desc: step.description,
                richTextParams: step.richTextParams,
                isSelected: step.id == selected,
                isEditable: step.isEditable,
              );
```

This way we don't have this condition repeated: `titleInDetails != null ? titleInDetails! : title,`.

File: catalyst_voices/lib/pages/workspace/rich_text/problem_statement.dart (hunk 3 above)
Comment (digitalheartxs) [reply catalyst_voices/lib/pages/workspace/rich_text/problem_statement.dart:3]: tbh, I can't see value in doing that. Especially that's only temporary solution.

**Review by dtscalac — APPROVED** [review]
LGTM

**Pull request merged**